ros2_generate_interface_docs/ros2_generate_interface_docs/utils.py

```diff
@@ -0,0 +1,452 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import array
+import errno
+import html
+from io import StringIO
+import os
+import sys
+
+import em
+
+from rosidl_parser.definition import AbstractNestedType, BoundedSequence, NamespacedType
+
+from rosidl_runtime_py import get_interface_path
+
+
+_TEMPLATES_DIR = 'templates'
+IGNORED_KEYS = ['__slots__', '__doc__', '_fields_and_field_types', 'SLOT_TYPES', '__init__',
+                '__repr__', '__eq__', '__hash__', 'get_fields_and_field_types', '__module__']
+
+
+def generate_raw_text(raw_text):
+    """
+    Generate the text HTML for the message fields.
+
+    Lines that starts with '#' (comments) will be shown in blue, otherwise black.
+
+    :param raw_text: content of the interface
+    :type raw_text: str
+    :returns: string with the generated HTML
+    :rtype: str
+    """
+    raw_str = ''
+    for line in raw_text.splitlines():
+        parts = line.split('#')
+        if len(parts) > 1:
+            raw_str = raw_str + parts[0] \
+                + '<a style="color:blue">#%s</a></br>\n' % ('#'.join(parts[1:]))
+        else:
+            raw_str = raw_str + '%s<br/>\n' % parts[0]
+    return html.escape(raw_str)
```

Comment:
Doesn't this escape the link tag from above?

Reply:
I unescaped them in the template

Reply:
That is just incorrect since that would also unescape escaped e.g. `&` from within comments.

Reply:
I think I fixed this with this commit (moved this part to the template) f1b227c